core/keyserver/revisions.go

```diff
@@ -133,17 +133,18 @@ func (s *Server) ListMutations(ctx context.Context, in *pb.ListMutationsRequest)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "ReadBatch(%v, %v): %v", in.DirectoryId, in.Revision, err)
 	}
-	rt, err := SourceMap(meta.Sources).ParseToken(in.PageToken)
+	rt, err := SourceList(meta.Sources).ParseToken(in.PageToken)
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "Failed parsing page_token: %v: %v", in.PageToken, err)
 	}
 
 	// Read PageSize + 1 messages from the log to see if there is another page.
 	high := meta.Sources[rt.ShardId].HighestWatermark
-	msgs, err := s.logs.ReadLog(ctx, d.DirectoryID, rt.ShardId, rt.LowWatermark, high, in.PageSize+1)
+	logID := meta.Sources[rt.ShardId].LogId
```

Comment:
`ShardId` vs. `LogId` looks confusing to me. Is `ShardId` just meant to be index in the list of `Slices` that `meta` contains? So this ID makes sense only in combination with a single `meta` message. I'd recommend using a less "global" name for it then. Maybe `SliceIndex`? Although I realize that almost any name here can be confusing.

Reply:
Renamed to `SliceIndex`